### PR TITLE
Backport fix Issue 20441 - Wrong code -O -fPIC and pointer subtraction

### PR DESCRIPTION
**Pull request by don-clugston-sociomantic**
Issue 20441 - Wrong code with -O -fPIC and pointer subtraction

### CODE REVIEW

**Comment by daniel-zullo** [comment]
the file `src/dmd/backend/cod2.d` does not exist. Should the file `src/dmd/backend/cod2.c` need be patched instead?

**Comment by don-clugston-sociomantic** [comment]
Yeah, I'm still working on this. Why on earth did Mihails do it this way? It's totally crazy!

**Comment by don-clugston-sociomantic** [comment]
It passes! Wow that was hard!

File: patches/dmd/0004-fix-Issue-20441-Wrong-code-with-O-fPIC-and-pointer-s.patch [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+diff --git a/src/dmd/backend/cod2.c b/src/dmd/backend/cod2.c
+index eecbba919..107127bbc 100755
+--- a/src/dmd/backend/cod2.c
++++ b/src/dmd/backend/cod2.c
+@@ -690,6 +690,8 @@ void cdorth(CodeBuilder& cdb,elem *e,regm_t *pretregs)
+         break;
+ 
+     case OPrelconst:
++        if (I64 && (config.flags3 & CFG3pic || config.exe == EX_WIN64))
++                goto L2;
```
Comment (daniel-zullo) [inline patches/dmd/0004-fix-Issue-20441-Wrong-code-with-O-fPIC-and-pointer-s.patch:21]: it falls through the same code than `Default`, right?

**Review by daniel-zullo — APPROVED** [review]
I've just have a question and LGTM

File: patches/dmd/0004-fix-Issue-20441-Wrong-code-with-O-fPIC-and-pointer-s.patch (hunk 1 above)
Comment (don-clugston-sociomantic) [reply patches/dmd/0004-fix-Issue-20441-Wrong-code-with-O-fPIC-and-pointer-s.patch:21]: Yes. The code is:
```
   case OPind:                                 /* if addressing mode   */
        if (!e2->Ecount)                        /* if not CSE           */
                goto L1;                        /* try OP reg,EA        */
        /* FALL-THROUGH */
    default:                                    /* operator node        */
    L2:
```
there is no `goto default` in C, so it needs to be `goto L2`.

**Review by daniel-zullo — APPROVED** [review]
LGTM

**Pull request merged**